queries/leo/highlights.scm

```diff
@@ -0,0 +1,264 @@
+((identifier) @variable (#set! "priority" 95))
+
+[
+ "assert"
+ "assert_eq"
+ "assert_neq"
+ "block"
+ "console"
+ "const"
+ "constant"
+ "finalize"
+ "in"
+ "let"
+ "mapping"
+ "private"
+ "program"
+ "public"
+ "record"
+ "self"
+ "struct"
+] @keyword
+
+[
+ "transition"
+ "function"
+ "inline"
+] @keyword.function
+
+"import" @include
+
+"return" @keyword.return
+
+"for" @repeat
+
+[ 
+  "if"
+  "then"
+  "else"
+] @conditional
+
+[ ";" "," "::"] @punctuation.delimiter
+
+[
+"!"
+
+"&&"
+"||"
+
+"=="
+"!="
+
+"<"
+"<="
+">"
+">="
+
+"&"
+"|"
+"^"
+
+"<<"
+">>"
+
+"+"
+"-"
+"*"
+"/"
+"%"
+"**"
+
+"="
+
+"+="
+"-="
+"*="
+"/="
+"%="
+"**="
+
+"<<="
+">>="
+
+"&="
+"|="
+"^="
+
+"&&="
+"||="
+
+] @operator
+
+(comment) @comment
+
+(boolean_literal) @boolean
+
+(ternary_expression
+  (_)
+  (ternary_if) @conditional
+  (_)
+  (ternary_else) @conditional
+  (_))
+
+(constant_declaration 
+  (identifier) @constant
+)
+
+;program name if with right extension "aleo" -> okay
+(program_declaration
+  (program_id 
+    name: (identifier) 
+    extension: (identifier) @ext (#eq? @ext "aleo")) @string
+)
+
+;program name if with not with extension "aleo" -> error
+(program_declaration
+  (program_id 
+    name: (identifier) 
+    extension: (identifier) @ext (#not-eq? @ext "aleo")) @punctuation.delimiter
+)
+
+;record declaration
+(record_declaration (identifier) @structure) 
+
+;struct component 
+(struct_component_declaration 
+  (identifier) @parameter
+  (type)       @type
+) 
+
+(type) @type
+
+(associated_constant) @constant
+
+(self_caller) @constant
+
+(block_height) @constant
+
+(transition_declaration
+  .
+  name: (identifier) @function.bultin
+  .
+  (function_parameters)*
+  .
+  (return_arrow) @keyword.return
+  .
+  (type)
+  .
+  (block)
+  .
+)
+
+;external transition call
+;will be wrong for internal function calls!
+(free_function_call
+  (identifier)  @function ) 
+
+(function_declaration
+  .
+  name: (identifier) @function
+  .
+  (function_parameters)*
+  .
+  (return_arrow) @keyword.return
+  .
+  (type)
+  .
+  (block)
+  .
+)
+
+(inline_declaration
+  .
+  name: (identifier) @function
+  .
+  (function_parameters)*
+  .
+  (return_arrow) @keyword.return
+  .
+  (type)
+  .
+  (block)
+  .
+)
+
+(method_call
+  (_)
+  (identifier) @method
+)
+
+(function_parameter
+ (identifier) @parameter
+) 
+
+(struct_declaration
+  name: (identifier) @structure)
+
+(unsigned_literal) @number
+
+(signed_literal) @number
+
+(field_literal) @number
+
+(product_group_literal) @number
+
+(affine_group_literal) @number
+
+(scalar_literal) @number
+
+(boolean_literal) @boolean
+
+(address_literal) @number
```

Comment:
```suggestion
[
  (unsigned_literal)
  (signed_literal)
  ....
] @number

Reply:
Fixed: [91454aae](https://github.com/r001/nvim-treesitter/commit/91454aae7c2f82c6d804167da8a8d2dab2c0fdfb)

Reply:
All these `fixed` commits aren't on github just yet so I suppose I'll wait till it's up to review again